### ISSUE TITLE
New-CmAzIaasVm : move $virtualMachine.fullName Get-CmAzResourceName order 

### ISSUE THREAD

**Issue by JBiasilloTM** [Trustmarque/cloudmarque-azure-pwsh#216]
move $virtualMachine.fullName 1st so that nic and os disk match fullname (fullname was last so nic os didnt have 'vm' prefix)

**Comment by nathankitchen** [comment]
The default convention is based on the [Microsoft](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations) [recommendations](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming) as part of the CAF framework.

The naming convention is defined on a per-project basis so you're free to update `/_names/resources.yml` to change the `NetworkInterfaceCard` to use a different generator, and set up the generator to have whatever naming convention order you see fit.

That aside, I think it makes sense to make this the default for future projects, so we'll queue this up for the next release. Change required is:

In `/_names/resources.yml`:
``` yaml
- name: NetworkInterfaceCard
  shortname: nic
  minLength: 1
  maxLength: 64
  generator: standard # change this to 'nic'
```

In `/_names/generators.yml` add the following generator:
``` yaml
  - name: nic
    separator: "-"
    case: lower
    components:
      - source: name    # may need code review to check what this is set to in context of VM deployment
      - source: resource
      - source: environment
      - source: region
      - source: hash
```

**Comment by JBiasilloTM** [comment]
The naming issue just about about the nic- prefix. Nic using computer name it could have a different name the the VM fullname visible for the VM via Portal etc. 

**current order**
$virtualMachine.computerName = Get-CmAzResourceName -Resource "ComputerName" -Architecture "Core" -Region $virtualMachine.location -Name $virtualMachine.name -MaxLength 15
					$virtualMachine.nicName = Get-CmAzResourceName -Resource "NetworkInterfaceCard" -Architecture "IaaS" -Region $virtualMachine.location -Name $virtualMachine.name
					$virtualMachine.osDisk.Name = Get-CmAzResourceName -Resource "OSDisk" -Architecture "IaaS" -Region $virtualMachine.location -Name $virtualMachine.name
					$virtualMachine.fullName = Get-CmAzResourceName -Resource "VirtualMachine" -Architecture "IaaS" -Region $virtualMachine.location -Name $virtualMachine.name

**extensions order using vm fullname**
					$virtualMachine.fullName = Get-CmAzResourceName -Resource "VirtualMachine" -Architecture "IaaS" -Region $SettingsObject.location -Name $virtualMachine.name
					$virtualMachine.computerName = Get-CmAzResourceName -Resource "ComputerName" -Architecture "Core" -Region $SettingsObject.location -Name $virtualMachine.name -MaxLength 15
					$virtualMachine.nicName = Get-CmAzResourceName -Resource "NetworkInterfaceCard" -Architecture "IaaS" -Region $SettingsObject.location -Name $virtualMachine.fullName
					$virtualMachine.osDisk.Name = Get-CmAzResourceName -Resource "OSDisk" -Architecture "IaaS" -Region $SettingsObject.location -Name $virtualMachine.fullName

**Comment by tm-luke** [comment]
Fixed  in 1.0.10
